cdragontoolbox/binfile.py

```diff
@@ -361,6 +417,30 @@ def read_binfile_entry(self, htype):
         assert self.f.tell() - pos == length
         return entry
 
+    def read_patch_section(self):
+        """Reads the entire PTCH section"""
+
+        count = self.read_u32()
+        patch_entries = {}
+        for _ in range(count):
+            hpath = self.read_fmt('<2L')[0]
+            btype = self.parse_bintype(self.read_u8())
+            objectpath = self.read_string()
+            parts = objectpath.split('.')
+            binvalue = self.read_field_basic(compute_binhash(parts[-1]), btype)
+            if hpath not in patch_entries:
+                patch_entries[hpath] = BinPtchEntry(hpath, BinNested([]))
+
+            current_nesting = patch_entries[hpath].value
+            for i in range(0, len(parts)-1):
+                if parts[i] not in current_nesting:
+                    current_nesting[parts[i]] = BinNestedField(compute_binhash(parts[i]), BinNested([]))
+                current_nesting = current_nesting.getv(parts[i])
```

Comment:
You can iterate directly on parts using `for part in parts[:-1]:` then using `part` instead of `parts[i]`.
(I included that in my previous snippet but forgot to change the `for`. I Should have explicit that additional change. :( )